server/src/jump_definition.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) ZeroC, Inc.
+
+use slicec::{
+    compilation_state::CompilationState,
+    grammar::{
+        Class, Commentable, CustomType, Entity, Enum, Enumerator, Exception, Field, Identifier,
+        Interface, MessageComponent, Module, NamedSymbol, Operation, Parameter, Struct, TypeAlias,
+        TypeRef, TypeRefDefinition, Types,
+    },
+    slice_file::{Location, SliceFile, Span},
+    visitor::Visitor,
+};
+use tower_lsp::lsp_types::{Position, Url};
+
+pub fn get_definition_span(state: &CompilationState, uri: Url, position: Position) -> Option<Span> {
+    // Attempt to convert the URL to a file path and then to a string
+    let file_path = uri.to_file_path().ok()?.to_str()?.to_owned();
+
+    // Attempt to retrieve the file from the state
+    let file = state.files.get(&file_path)?;
+
+    // Convert position to row and column to 1 based
+    let col = (position.character + 1) as usize;
+    let row = (position.line + 1) as usize;
+    let location: slicec::slice_file::Location = (row, col).into();
```

Comment:
Does it work without this type hint? I suspect it can infer it here. But who knows, maybe it can't,
```suggestion
    let location = (row, col).into();
```